foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,227 @@
+            root (:py:class:`~selenium.webdriver.remote.webelement.WebElement`): WebDriver element object for the bookmark panel
+
+        Returns:
+            :py:class:`BaseBookmark`: Bookmark instance or None
+        """
+        with window.selenium.context(window.selenium.CONTEXT_CHROME):
+            try:
+                return BasicBookmark(window, root)
+            except NoSuchElementException:
+                return None
+
+    @property
+    def is_bookmarked(self) -> bool:
+        """Checks if the current page is bookmarked.
+
+        Returns:
+            bool: True if the page is bookmarked, False otherwise.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            star_button_image = self.find_element(BookmarkLocators.STAR_BUTTON_IMAGE)
+            if star_button_image is not None:
+                return star_button_image.get_attribute("starred") == "true"
+            else:
+                return False
+
+    def add(self) -> None:
+        """Add a Bookmark using the star button."""
+        self.click_element(BookmarkLocators.STAR_BUTTON)
+        self.click_element(BookmarkLocators.FOLDER_MENU)
+        self.click_element(BookmarkLocators.OTHER_BOOKMARKS_STAR)
+        self.click_element(BookmarkLocators.SAVE_BUTTON)
+
+    def retrieve_bookmark(self, label: str) -> bool:
+        """
+        Check if a bookmark with the given label exists under 'Other Bookmarks'.
+
+        Args:
+            label (str): The name of the bookmark to search for.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.open_bookmark_menu(
+                locator_panel=BookmarkLocators.PANEL_MENU,
+                locator_bookmark=BookmarkLocators.PANEL_BOOKMARK_MENU,
+            )
+            panel_bookmarks = self.find_element(BookmarkLocators.PANEL_BOOKMARK_TOOLBAR)
+            if panel_bookmarks is None:
+                return False
+            menu_items = panel_bookmarks.find_elements(
+                By.CSS_SELECTOR, "toolbarbutton.bookmark-item"
+            )
+            if not menu_items:
+                return False
+            for item in menu_items:
+                item_label = item.get_attribute("label")
+                if item_label and label.lower() in item_label.lower():
+                    return True
+
+            return False
+
+    def delete(self) -> None:
+        """Delete a bookmark using the star button."""
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            star_button_image = self.find_element(BookmarkLocators.STAR_BUTTON_IMAGE)
+            if star_button_image and star_button_image.get_attribute("starred") == "true":
+                self.click_element(BookmarkLocators.STAR_BUTTON)
+                self.click_element(BookmarkLocators.REMOVE_BUTTON)
+
+
+class AdvancedBookmark(BasicBookmark):
+    """Handles advanced bookmark operations."""
+
```

Comment:
What's an "Advanced Bookmark" exactly?

Reply:
Okay, so in Desktop Smoke test, a similar term is used so I decided to adopt it in FoxPuppet. Basically, an advanced bookmark is a more detailed way of adding a bookmark, which involves `name`, `url`, `tags`, and `keywords`, as against just simply using the `url` alone when creating a `Basic Bookmark` with the `star button`.

Reply:
Okay let me ask Ben C for clarifications

Reply:
Here is a link to it: https://github.com/mozilla/fx-desktop-qa-automation/blob/65fe54ea0badf6d665e55b4649da32c8a10b4598/modules/browser_object_navigation.py#L206

Reply:
@ben-c-at-moz Can you explain what your "Advanced Bookmark" action does exactly?

Reply:
Yeah, I agree, the name is a little confusing.

Reply:
But I'm fine if we just want to keep using it for more detailed bookmark operations, as long as we explain it.

Reply:
Could we just call it `Bookmarks` and put all the functionality into one class? If a user wants to do it through the "Other Bookmarks" option, we can just have a flag or something when they create the bookmark.